Documentation/Design/Design Doc Template.md

```diff
@@ -0,0 +1,151 @@
+# Feature Name
+
+## Feature Overview
+
+A description of the feature.
+
+
+## Commands
+
+### `command 1`
+
+`command 1 syntax`
+
+A brief description of the command, including any optional parameters or parameters with default values.
+
+### `command 2`
+
+`command 2 syntax`
+
+A brief description of the command, including any optional parameters or parameters with default values.
+
+## Usage
+
+```gls
+command 1 examples
+```
+
+```gls
+command 2 examples
```

Comment:
Seems good. I'm wondering if it's possible to include a spec for optional/repeating parameters here, or if that should be somewhere else... Thoughts?